src/outputgeneration/ParseTreeWriter.js

```diff
@@ -177,6 +181,10 @@ export class ParseTreeWriter extends ParseTreeVisitor {
   visitArrayPattern(tree) {
     this.write_(OPEN_SQUARE);
     this.writeList_(tree.elements, COMMA, false);
+    if (tree.elements[tree.elements.length - 1] === null) {
+      this.write_(COMMA);
+      this.writeSpace_();
```

Comment:
Seems like we could have a test that verifies this case (and fails without this code).